crates/bevy_ecs/src/reflect.rs

```diff
@@ -106,11 +112,22 @@ impl ReflectComponent {
     }
 
     /// Gets the value of this [`Component`] type from the entity as a reflected reference.
+    ///
+    /// If repeatedly fetching multiple components from the same entity, prefer using
+    /// [`ReflectComponent::reflect_ref`] where possible to avoid multiple entity location lookups.
     pub fn reflect<'a>(&self, world: &'a World, entity: Entity) -> Option<&'a dyn Reflect> {
         (self.0.reflect)(world, entity)
```

Comment:
Could this be changed to use the same function pointer as `reflect_ref`?

Reply:
That would require returning a reference bound to the lifetime of the dropped EntityRef.

Reply:
Apparently that is only true for the mutable variant -- the read-only version is bound to the lifetime of the `World`, not the `EntityRef`. Not entirely necessary though, I suppose.